### PR TITLE
Improve code coverage for CurlClient.

### PR DESCRIPTION
**Pull request by coryan**
I thought the existing tests would cover the libcurl failure cases,
obviously I was wrong. This reuses the tests for credential failures,
because otherwise we would need to rewrite basically the same tests.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1744)

### CODE REVIEW

File: google/cloud/storage/internal/curl_client_test.cc [hunk 1]
```diff
@@ -46,306 +49,358 @@ class FailingCredentials : public Credentials {
...
-    client_ = CurlClient::Create(std::make_shared<FailingCredentials>());
+  CurlClientTest() : endpoint_("CLOUD_STORAGE_TESTBENCH_ENDPOINT") {}
+
+  void SetUp() override {
+    std::string const error_type = GetParam();
+    if (error_type == "credentials-failure") {
+      client_ = CurlClient::Create(std::make_shared<FailingCredentials>());
+      expected_status_code_ = STATUS_ERROR_CODE;
+      expected_status_substr_ = STATUS_ERROR_MSG;
+    } else {
```
Comment (devbww) [inline google/cloud/storage/internal/curl_client_test.cc]: Should we check for "libcurl-failure"?

File: google/cloud/storage/internal/curl_client_test.cc [hunk 2]
```diff
@@ -46,306 +49,358 @@ class FailingCredentials : public Credentials {
...
+  }
 
-void TestCorrectFailureStatus(Status const& status) {
-  EXPECT_EQ(status.status_code(), STATUS_ERROR_CODE);
-  EXPECT_EQ(status.error_message(), STATUS_ERROR_MSG);
-}
+  std::shared_ptr<CurlClient> client_;
+  long expected_status_code_;
+  std::string expected_status_substr_;
+  testing_util::EnvironmentVariableRestore endpoint_;
```
Comment (devbww) [inline google/cloud/storage/internal/curl_client_test.cc]: endpoint_.TearDown() in (a new) ~CurlClientTest()?

File: google/cloud/storage/internal/curl_client_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/curl_client_test.cc]: Good idea, done.

File: google/cloud/storage/internal/curl_client_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/internal/curl_client_test.cc]: Indeed, for some reason I thought `endpoint_` had a destructor that did that, thanks for checking.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**